From c5228acc9e725f5bf478ee2e4452cf8dd45096fd Mon Sep 17 00:00:00 2001
From: Daniel Micay <danielmicay@gmail.com>
Date: Thu, 14 May 2015 19:14:53 -0400
Subject: [PATCH] enable secure_delete by default

This sanitizes deleted data by overwriting it with zeroes, rather than
having it persist within SQLite's free list. In some cases, the data may
end up persisting on the underlying storage due to wear levelling but
only a process with direct access to block devices would be able to
access it.

This can be disabled via `PRAGMA SECURE_DELETE = 0;` for niches where
deletion throughput is more important than enhanced privacy. The
performance impact is very small unless large strings and blobs are
being deleted. Android itself doesn't have any uses where disabling it
would make sense, and very few apps will be negatively impacted. As a
data point, Chromium and Firefox already use this for their own
databases.

Change-Id: I07f0e1584f0b2405f080198ad7b1f4e7ed7f68fb
---
 dist/Android.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dist/Android.mk b/dist/Android.mk
index 02c2409..20a7bb5 100644
--- a/dist/Android.mk
+++ b/dist/Android.mk
@@ -27,7 +27,8 @@ common_sqlite_flags := \
 	-DSQLITE_OMIT_BUILTIN_TEST \
 	-DSQLITE_OMIT_COMPILEOPTION_DIAGS \
 	-DSQLITE_OMIT_LOAD_EXTENSION \
-	-DSQLITE_DEFAULT_FILE_PERMISSIONS=0600
+	-DSQLITE_DEFAULT_FILE_PERMISSIONS=0600 \
+	-DSQLITE_SECURE_DELETE
 
 device_sqlite_flags := $(common_sqlite_flags) \
     -DSQLITE_ENABLE_ICU \
-- 
2.11.0

